src/value.h

```diff
@@ -20,57 +20,45 @@
 typedef enum {
   T_NULL = 0,
   T_STRING = 0x001,
-  T_INT32 = 0x002,
+  // T_INT32 = 0x002, // unused
   T_INT64 = 0x004,
-  T_UINT = 0x008,
-  T_BOOL = 0x010,
-  T_FLOAT = 0x020,
+  // T_UINT = 0x008, // unused
+  T_BOOL = 0x010, // Instance of T_INT64
+  // T_FLOAT = 0x020, // unused
   T_DOUBLE = 0x040,
   T_PTR = 0x080,
   T_CONSTSTRING = 0x100,
-
-  // special types for +inf and -inf on all types:
-  T_INF = 0x200,
-  T_NEGINF = 0x400,
-
 } SIType;
 
 #define SI_STRING (T_STRING | T_CONSTSTRING)
-#define SI_NUMERIC (T_INT32 | T_INT64 | T_UINT | T_FLOAT | T_DOUBLE)
+#define SI_NUMERIC (T_INT64 | T_DOUBLE)
 #define SI_TYPE(value) (value).type
 
-/* Returns true if aVal and bVal are of the same type, are both string types, or are both numeric types. */
-#define SI_COMPARABLE(aVal, bVal) ((aVal).type == (bVal).type || \
-		(((aVal).type & SI_NUMERIC) && (bVal).type & SI_NUMERIC) || \
-		(((aVal).type & SI_STRING) && (bVal).type & SI_STRING))
+/* Retrieve the numeric associated with an SIValue without explicitly
+ * assigning it a type. */
+#define SI_GET_NUMERIC(v) ((v).type == T_DOUBLE ? (v).doubleval : (v).longval)
```

Comment:
Do we want this macro to validate if v is indeed numeric?

Reply:
I think that wouldn't be great, since we'd need to introduce another macro argument to indicate success or return a placeholder (`INT64_MAX`, probably).

Either way would mess up the current calling strategy, which shifts type resolution to the compiler. For example, subtraction is:
`SI_DoubleVal(SI_GET_NUMERIC(a) + SI_GET_NUMERIC(b));`

Also, of course, this macro can get called a lot, so I want it to be as tight as possible.